### PR TITLE
`Plagiarism checks`: Display text submissions even though similarity highlighting does not work

### PR DESCRIPTION
**Pull request by akesfeden**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

We had a case when server returned PlagiarismComparison matches which has `startA`/`startB` indices greater than the length of the length of `submissionA.elements`/`submissionB.elements`. In this and sileven though similarity highlighting does not work, the users should be able to see both text submissions. This allows them to at least compare the submissions manually.

### Description

This PR adds a fail-soft fix that will display text submissions even though matches cannot be mapped to elements correctly due to out-of-bounds indices.

Also, this PR adds "overall similarity" percentage above the split view for both instructors and students.

Additionally, resolves https://github.com/ls1intum/Artemis/issues/5403 with a new button for navigating to Exercise Plagiarism Detection page from the Plagiarism Case Instructor Detail view.

Finally, this PR addresses the root cause of out-of-bounds indices, which was caused by the swap operation between `submissionA` and `submissionB` of `PlagiarismComparison`. It did not swap `PlagiarismComparison.matches[].startA` and `startB` fields so the indices did not match.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text Exercise

1. Log in to Artemis
2. Navigate to Course Administration
3. Make two similar text submissions with both students
  To reproduce the problematic "Before" case in the screenshot:
    - One submission must be longer than the other.
    - There should be similarities starting near the end of the submissions (You can just copy/paste a short paragraph at the end of both submissions).
5. As the instructor, set the due date of the exercise to now and run plagiarism check from Exercise Detail page > Check Plagiarism page.
6. Notify a student about the plagiarism.
7. With both the instructor and the notified student, check if the exercise detail page shows both submission in the split view and highlighted similarities are correct.
8. Repeat the same steps for modeling and programming exercises.

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| plagiarism-case-instructor-detail-view.component.ts | 72.91% | 88.23% |
| plagiarism-split-view.component.ts | 90.32% | 94.91% |
| text-submission-viewer.component.ts | 79.31% | 94.59% |

### Screenshots

Before:
image

After handling out-of-bounds indices:
image

After fixing submission swapping:
image

New button for navigating to Exercise Plagiarism Detection page:
image

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Code changes and screenshots look good

File: src/main/webapp/i18n/de/plagiarism.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
                 "exercise": "Betroffene Übung",
                 "comparisons": "Vergleiche",
                 "comparison": "Vergleich",
+                "overallSimilarity": "Allgemeine Ähnlichkeit",
```
Comment (Hialus) [inline src/main/webapp/i18n/de/plagiarism.json:74]: I think "Gesamte Ähnlichkeit" or "Gesamtähnlichkeit" would fit better.

**Review by Hialus — APPROVED** [review]
Code changes look good. I just left one optional remark about the german translation.

File: src/main/webapp/i18n/de/plagiarism.json (hunk 1 above)
Comment (akesfeden) [reply src/main/webapp/i18n/de/plagiarism.json]: Thanks @Hialus, I applied your suggestion.

File: src/main/webapp/app/course/plagiarism-cases/shared/review/plagiarism-case-review.component.html [hunk 2]
```diff
@@ -1,7 +1,14 @@
 <ul ngbNav #nav="ngbNav" class="nav-tabs row" [destroyOnHide]="false">
     <ng-container *ngFor="let submission of plagiarismCase.plagiarismSubmissions; let i = index">
+        <div class="mt-2">
+            {{ 'artemisApp.plagiarism.plagiarismCases.overallSimilarity' | artemisTranslate }}: <strong>{{ submission.plagiarismComparison.similarity?.toFixed(1) }}%</strong>
+        </div>
         <li [ngbNavItem]="i" class="col px-0">
             <a ngbNavLink class="btn-sm text-dark text-center">{{ 'artemisApp.plagiarism.plagiarismCases.comparison' | artemisTranslate }} {{ i + 1 }}</a>
+            <div class="my-2" style="font-size: large; text-align: center">
```
Comment (bassner) [inline src/main/webapp/app/course/plagiarism-cases/shared/review/plagiarism-case-review.component.html:8]: Please do not use inline styles; either use bootstrap classes or define custom class an style the element in a ``scss`` file.

**Review by bassner — CHANGES_REQUESTED** [review]
Tested on staging like described, the highlighting and split view seems to work fine 👍

_Some issues:_

The new overall similarity box is displayed, but why twice? 
grafik

The headline above it is not well visible in dark mode:
grafik

We should fix these issues before merging!

Additionally, I have one code comment below.

File: src/main/webapp/app/course/plagiarism-cases/shared/review/plagiarism-case-review.component.html (hunk 2 above)
Comment (akesfeden) [reply src/main/webapp/app/course/plagiarism-cases/shared/review/plagiarism-case-review.component.html]: Thanks, I addressed the issues (duplicate similarity, dark text, inline styles).

File: src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts [hunk 3]
```diff
@@ -137,6 +137,23 @@ export class PlagiarismCaseInstructorDetailViewComponent implements OnInit, OnDe
...
+     * @param exerciseType type of exercise
+     */
+    getExerciseUrlSegment(exerciseType?: ExerciseType) {
+        switch (exerciseType) {
+            case ExerciseType.TEXT:
+                return 'text-exercises';
+            case ExerciseType.MODELING:
+                return 'modeling-exercises';
+            case ExerciseType.PROGRAMMING:
+                return 'programming-exercises/';
```
Comment (ge65cer) [inline src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts:151]: Unneeded forward slash?
Do we not have a common method for this somewhere?

Comment (akesfeden) [reply src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts:151]: Yes I'll remove the forward slash, thanks 👍 I searched for a common method, there are similar ones but they are also not generic enough.

Comment (akesfeden) [reply src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts]: I removed it in https://github.com/ls1intum/Artemis/pull/5544/commits/8352e047a809c951c4daed8fda0f3d3602a403d7

**Review by bassner — APPROVED** [review]
Thank you for addressing the issues. Looks good to me now! 👍

**Review by JohannesStoehr — APPROVED** [review]
Code looks good after the code changes

**Review by Santia-go — APPROVED** [review]
Tested on TS1
Working as expected. 
Additionally to [#5403](https://github.com/ls1intum/Artemis/issues/5403) you are closing also [#5537](https://github.com/ls1intum/Artemis/issues/5537) and [#5538](https://github.com/ls1intum/Artemis/issues/5538)

Thank you!

**Review by ge65cer — APPROVED** [review]
Thanks for the changes/answers.

**Review by Hialus — APPROVED** [review]
Code looks good.

**Pull request merged**